libs/wingsdk/scripts/generate-macros.mts

```diff
@@ -0,0 +1,45 @@
+import { MethodDeclaration, Project, SyntaxKind } from "ts-morph";
+
+const MACRO = "macro";
+
+const project = new Project();
+
+const sourceFiles = project.addSourceFilesAtPaths("src/std/*.ts");
+
+function extractJSDocs(node: MethodDeclaration) {
+  const jsDocs = node.getChildrenOfKind(SyntaxKind.JSDoc);
+  if (!jsDocs || !jsDocs.length) {
+    return undefined;
+  }
+  return jsDocs[0]
+    .getTags()
+    .find((d) => d.getTagName() === MACRO)
+    ?.getComment();
+}
+
+let macroFileContent = `
+// This file is generated by the generate-macro npm script. Changes will be overridden.
+`;
+
+for (const file of sourceFiles) {
+  for (const cls of file.getClasses()) {
+    for (const mtd of cls.getMethods()) {
+      const macro = extractJSDocs(mtd);
+      if (macro) {
+        const spreadArgs = macro.includes("...$args$") ? "..." : "";
+        macroFileContent += `
+        exports.__${cls.getName()}_${mtd.getName()} = (skipIfNil, $self$, ${spreadArgs}$args$) => {
+          if (skipIfNil && $self$ === undefined) return $self$;
+          return ${macro}
+        }
+        `;
+      }
+    }
+  }
+}
+
+project.createSourceFile("lib/macros.js", macroFileContent, {
```

Comment:
Is it possible to generate it as a typescript file inside `src/` so that the file will be saved to git? If there are type checking issues, maybe they can be ignored by generating `// @ts-ignore` or by adding the file name to the excludes section in `tsconfig.json`.

Reply:
it can create a js file and save it as a ts, there are plenty of unsolvable (tried!) type-checking issues, and the file itself- since it's actually js, and un-linted, looks very ugly :( 
Do you wish to commit it anyway? for verification?

Reply:
I see - let's try leaving it as is then